packages/turbo-telemetry/src/client.test.ts

```diff
@@ -159,33 +211,58 @@ describe("TelemetryClient", () => {
       command: "test-command",
       status: "start",
     });
-    expect(got.post).not.toHaveBeenCalled();
+
+    assert.equal(mockPost.mock.callCount(), 0);
 
     await client.close();
 
-    expect(got.post).toHaveBeenCalled();
-    expect(got.post).toHaveBeenCalledWith(
-      "https://example.com/api/turborepo/v1/events",
-      expect.objectContaining({
-        json: [
-          {
-            package: {
-              id: expect.any(String) as string,
-              key: "command:test-command",
-              value: "start",
-              package_name: "create-turbo",
-              package_version: "1.0.0",
-            },
-          },
-        ],
-        headers: {
-          "User-Agent": expect.stringContaining("create-turbo 1.0.0") as string,
-          "x-turbo-session-id": expect.any(String) as string,
-          "x-turbo-telemetry-id": "telemetry-test-id",
-        },
-      })
-    );
-
-    expect(client.hasPendingEvents()).toBe(false);
+    assert.equal(mockPost.mock.callCount(), 1);
+    assert.equal(
+      mockPost.mock.calls[0].arguments[0],
+      "https://example.com/api/turborepo/v1/events"
+    );
+
+    assert.equal(mockPost.mock.calls[0].arguments[1].json.length, 1);
+    assert.deepEqual(
+      Object.keys(mockPost.mock.calls[0].arguments[1].json[0].package),
+      ["id", "key", "value", "package_name", "package_version", "parent_id"]
+    );
+    assert.equal(
+      typeof mockPost.mock.calls[0].arguments[1].json[0].package.id,
+      "string"
+    );
+    assert.equal(
+      mockPost.mock.calls[0].arguments[1].json[0].package.key,
+      "command:test-command"
+    );
+    assert.equal(
+      mockPost.mock.calls[0].arguments[1].json[0].package.value,
+      "start"
+    );
+    assert.equal(
+      mockPost.mock.calls[0].arguments[1].json[0].package.package_name,
+      "create-turbo"
+    );
+    assert.equal(
+      mockPost.mock.calls[0].arguments[1].json[0].package.package_version,
+      "1.0.0"
+    );
+
+    assert.equal(
+      "x-turbo-session-id" in mockPost.mock.calls[0].arguments[1].headers,
+      true
+    );
+    assert.equal(
+      "x-turbo-telemetry-id" in mockPost.mock.calls[0].arguments[1].headers,
+      true
+    );
+    assert.equal(
+      /create-turbo 1\.0\.0/.test(
+        mockPost.mock.calls[0].arguments[1].headers["User-Agent"]
+      ),
+      true
+    );
```

Comment:
I think [`assert.matches`](https://nodejs.org/api/assert.html#assertmatchstring-regexp-message) is more fitting
```suggestion
    assert.matches(
      mockPost.mock.calls[0].arguments[1].headers["User-Agent"],
      /create-turbo 1\.0\.0/
    );
```